DESIGN.md

```diff
@@ -0,0 +1,183 @@
+
+    1. requirements.txt
+    1. setup.cfg
+    1. setup.py
+    1. Pipfile
+    1. Pipfile.lock
+    1. pyproject.toml
+    1. environment.yml
+    1. yarn.lock
+    1. package.json
+    1. package-lock.json
+    1. DESCRIPTION
+
+    when repositories contain subdirectories with any of these dependency files, we need to do more nested/recursing evaluation
+
+1. for each of the project dependencies,
+    1. figure out whose copy is being used (e.g. PyPI, GitHub, crates.io, npm, maven, Anaconda cloud, bower, etc. more [here](https://en.wikipedia.org/wiki/List_of_software_package_management_systems#Application-level_package_managers))
+    1. identify the license as stated on the platform whose copy we're using
+1. yields a one-to-many graph (a tree) with relations ("If I have a GPL-3.0 here, can I have Apache-2.0 one level higher?")
+1. figure out if there is any license of a dependency that yields a conflict with the top level license (should be Apache-2.0) (based on which rules?)
+1. choose which result best represents the situation:
+    1. didn't find a dependency file
+    1. found a dependency file,
+        1. every name was resolvable
+            1. no conflicts
+            1. had conflicts (with list of conflicts found)
+        1. not every name was resolvable
+            1. needs human evaluation
+
+## High level implementation plan
+
+After [evaluating existing tools](https://github.com/tortellini-tools/action/issues/2) we decided to wrap [ort](https://github.com/oss-review-toolkit/ort) in a script. As ort can detect licenses of dependences of a freshly cloned repository and produce a nice HTML report and machine readable reports. Ort is also widely used and supported by the OSS community.
+
+1. Write a script to regularly
+    - Clone the repository
+    - For each repo generate a report using [ort](https://github.com/oss-review-toolkit/ort)
+2. Define curation files
+    - Start with some predefined curations
+    - If users want to change or overwrite these curations they can
+3. Add index page to list repos
+    - Each repo should have a detailed report (html webapp)
+4. Instructions for program managers and engineers
+    - How to run analysis if you are a program manager
+    - How to run analysis if you are an engineer
+    - How to update curations
+
+## Technical implementation details
+
+The tool will be written as a GitHub action so it can be called in a GitHub workflow
+The action has 2 modes:
+
+### List of repos
+
+Steps in a Github workflow:
+
+1. Weekly scheduled (eg [fairtally-test](https://github.com/jmaassen/fairtally-test/blob/main/.github/workflows/fairtally.yml))
+2. Action fetches repos from RSD
+3. Use ort Docker image from https://hub.docker.com/r/philipssoftware/ort/
+4. Run https://github.com/tortellini-tools/action/blob/rsd-software-vs-ort/ort/batch.sh, replace shell script with Typescript
```

Comment:
we could update this

Reply:
I dont want to adapt the plan/design to the implementation too much, so would like keep it.

Reply:
good with me